internal/validations/clusterlogging/validate_clusterlogging_spec.go

```diff
@@ -15,8 +21,16 @@ func validateClusterLoggingSpec(cl v1.ClusterLogging) error {
 		return nil
 	}
 	spec := cl.Spec
+
 	if spec.Forwarder != nil || spec.LogStore != nil || spec.Curation != nil || spec.Visualization != nil {
```

Comment:
I think we are loosing something here in validation of ClusterLogging.  We have 2 scenarios:
* legacy: you must define CLF and it must be named openshift-logging/instance and it can have any of these fields
* mCLF: it must ns/name match the CLF and ONLY define spec.collection

This change appears to break the legacy scenario. 

I believe the load of CL needs to occur before this IF check and then based upon the result, we check and validate to see which of the scenarios is valid

Reply:
I don't think that it will break the legacy scenario because we check this above:
```
  if cl.Namespace == constants.OpenshiftNS && cl.Name == constants.SingletonName {
		if cl.Spec.Collection != nil && !cl.Spec.Collection.Type.IsSupportedCollector() {
			return errors.NewValidationError("Collector implementation is not supported: %q", cl.Spec.Collection.Type)
		}
		return nil
	}
```